### ISSUE TITLE
Path format template function to match metadata to range

### ISSUE THREAD

**Issue by Kraymer** [beetbox/beets#740]
This one concerns the file naming scheme of the music library. Long time I got that in my mind, time to put it on _paper_.

Currently, template replace `$field` with the appropriate value from the track’s metadata.
The idea would be to introduce a new field prefixing symbol that would enable user to hint beets which metadata to consider to pick a directory amongst a list of candidates.
It would be especially to handle range formatted directories such as _'A-E'_ or _1980-90_ ... 

---

I guess my description is a little opaque, le'ts illustrate with an example.
Bob wants to import _Let it be_ album to this destination :  

```
 /Pop/70s/[A-C]/Beatles-Let_it_be-1970
```

Notably, Bob uses folder names to describe periods ranges (_60s, 70s, 80s, etc_) and group artists (_[A-C], [D-G], etc_).
It's possible to handle such structure in beets by defining template functions, eg  

```
  /$genre/%rangify_year($year)/%rangify_artist($albumartist)/$albumartist-$album-$year
```

However, because a vast number of ranges representations can be infered from a tag, the template functions (`rangify_year`, `rangify_artist`) cannot be generic and will map a string to the corresponding range folder name **specific to the user folders organization**. 

I'm suggesting having an additional abstraction layer, a syntax to auto-magically resolve such paths, ie it would become :dizzy:  : 

```
/$genre/%year/%albumartist/$albumartist-$album-$year
```

Here the symbol `%` followed by a metadata id means "find me the most suitable directory (at current location in the path) whose name is a bucket for metadata id".
If id3 year == _1970_ and children folders of _Pop_ are _60s, 70s, 80s_ `%year` should return _70s_.

Things to consider :  
1. contrary to `$field`, `%field` would only return **existing directories** (we need to restrict the search set!)
2. which implies that contrary to `$field`, `%field` must be passed at runtime the evaluation of the preceding path tokens (so that we can `ls` and get folders list). 

**tldr; I think there is a _possibility_ to facilitate description of ranges in paths formats. Would that help a lot of users?.. that I'd like to know.
By limiting ourself to detect existing folders, we could make something that _just works_. However I have no idea how to implement that efficiently (ideas are welcome).**
